### PR TITLE
Add first draft for lab 3 using Prefect

### LINKED ISSUE #66: Lab 3 - Check if notebook still works
There seems to be an issue with part 1 where the model can't be trained. A student reported this in their demo and report. Executing the same step on Azure is no problem, seems a version issue. Also check the Docker images versions.

See <https://github.com/HOGENT-MLOps/mlops-2324-ArvidVanDorpe1/blob/main/ml-workflow/03-Azure/Azure_done.ipynb>

### PR DESCRIPTION
**Pull request by thomasaelbrecht**
This PR is not meant to be merged immediately. I just wanted to hear your feedback on this first draft.

Should we give them a very simple pipeline to demonstrate the working of `@task` and `@flow` or should the be able to do it all themselves? In the lab assignment you'll see that I just point to some documentation and that they need to develop the entire pipeline themselves.

We could also give the a base structure to plug the code into.

Closes #66

### CODE REVIEW

File: assignments/03-ml-workflow.md [hunk 1]
```diff
@@ -7,6 +7,15 @@
...
+<!-- These criteria are checked: -->
+
+- Show that you've executed the notebook and pushed it to the repository
+  - Show that your Jupyter notebook contains all cells' output
+- Show that you wrote an elaborate lab report in Markdown and pushed it to the repository
+  - Show that it contains the answers to the questions in the lab assignment
+
+<!-- TODO: update criteria -->
+
 - Show that you created an Azure ML workspace
```
Comment (MartenBE) [inline assignments/03-ml-workflow.md:19]: Remove Azure

File: assignments/03-ml-workflow.md [hunk 2]
```diff
@@ -16,48 +25,120 @@
...
+```
+
+Finally install the required dependencies:
+
+```bash
+pip install -r requirements.txt
+```
+
+:question: Where are the dependencies installed?
+
```
Comment (MartenBE) [inline assignments/03-ml-workflow.md:84]: If we handle python venv here, then it can be omitted from the monitoring lab.

**Review by MartenBE — CHANGES_REQUESTED** [review]

**Comment by MartenBE** [comment]
I think it is OK to let them try to figure it out for themselves. We just need to be really clear about the miminum requirements we want to see. If it proves to difficult, we can always give hints or explanation during class hours and adjust it for 2025-2026.

File: assignments/03-ml-workflow.md (hunk 1 above)
Comment (thomasaelbrecht) [reply assignments/03-ml-workflow.md:19]: I'll remove these criteria when I have good alternatives for Prefect.

**Comment by thomasaelbrecht** [comment]
Model deployment is now not present in the assignment, since we work locally. Optionally we can provide a Node.js server to try to perform a prediction?

**Comment by MartenBE** [comment]
I have merged the PR, Adding the deployment could be nice, but if it is too much of a hassle, we could always leave it open as a possible extension

**Pull request merged**